Lombiq.Tests.UI/OrchardCoreUITestBase.cs

```diff
@@ -262,17 +262,18 @@ public abstract class OrchardCoreUITestBase
             BrowserConfiguration = { Browser = browser },
         };
 
-        ProgramCli.DefaultShellCliCommandFactory = OSDependentShellCliCommandFactory
-            .UseCmdForWindows()
-            .UseForOtherOS(new BashShellCliCommandFactory("-login"));
-
         configuration.SetupConfiguration.SetupOperation = setupOperation;
 
         if (changeConfigurationAsync != null) await changeConfigurationAsync(configuration);
 
         await UITestExecutor.ExecuteOrchardCoreTestAsync(testManifest, configuration);
     }
 
+    static OrchardCoreUITestBase() =>
+        ProgramCli.DefaultShellCliCommandFactory = OSDependentShellCliCommandFactory
+            .UseCmdForWindows()
+            .UseForOtherOS(new BashShellCliCommandFactory("-login"));
+
```

Comment:
We can't. Static constructors are called when the type is first referenced. So if we want to let the library consumer customize it in the `OrchardCoreUITestBase` overload as described in _Linux.md_, then we must be absolutely certain that this code has run before the setup customization. In general you should only use static constructor to set data that will be immediately interacted with. Every test calls to `OrchardCoreUITestBase` very early so that's perfect.
In this case I have verified that `AtataFactory`'s static constructor actually executes later than e.g. the configuration in `Lombiq.OSOCE.Tests.UI.UITestBase.ExecuteTestAsync`.

Reply:
Then at least encapsulate this into `AtataFactory` so we don't have basic Atata configuration in multiple places.